packages/masonry/README.md

```diff
@@ -0,0 +1,203 @@
+<p>
+  <img width="100%" src="https://assets.solidjs.com/banner?type=Primitives&background=tiles&project=masonry" alt="Solid Primitives masonry">
+</p>
+
+# @solid-primitives/masonry
+
+[![turborepo](https://img.shields.io/badge/built%20with-turborepo-cc00ff.svg?style=for-the-badge&logo=turborepo)](https://turborepo.org/)
+[![size](https://img.shields.io/bundlephobia/minzip/@solid-primitives/masonry?style=for-the-badge&label=size)](https://bundlephobia.com/package/@solid-primitives/masonry)
+[![version](https://img.shields.io/npm/v/@solid-primitives/masonry?style=for-the-badge)](https://www.npmjs.com/package/@solid-primitives/masonry)
+[![stage](https://img.shields.io/endpoint?style=for-the-badge&url=https%3A%2F%2Fraw.githubusercontent.com%2Fsolidjs-community%2Fsolid-primitives%2Fmain%2Fassets%2Fbadges%2Fstage-0.json)](https://github.com/solidjs-community/solid-primitives#contribution-process)
+
+Primitives for creating a reactive masonry layout.
+
+- [`createMasonry`](#createMasonry) - Creates a masonry layout calculation from a reactive source array.
+
+## Installation
+
+```bash
+npm install @solid-primitives/masonry
+# or
+yarn add @solid-primitives/masonry
+# or
+pnpm add @solid-primitives/masonry
+```
+
+## `createMasonry`
+
+Creates a masonry layout calculation from a reactive source array.
+
+It splits the items into columns and calculates the order based on the height of each item.
+
+The masonary is expected to be rendreded in a flex container with `flex-direction: column`, `flex-wrap: wrap` and limited height
+to force the items to wrap.
+
+## How to use it
+
+`createMasonry` requires an `options` object with the following properties:
+
+- `source` - Accessor returning the source array of items to be mapped.
+
+  When updating the array, the masonry will be recalculated.
+  The items are compared by reference.
+
+- `columns` - The number of columns to split the items into.
+
+  This can be an accessor to provide a reactive number of columns.
+
+- `mapHeight` - A function that maps the source item to a height.
+
+  This function is not reactive, it will be called only once for each item.
```

Comment:
> This is missing what measurement is expected (probably pixels).

it doesn't expect any specific unit, just a number, what it means is up to the user

> only further down in the text it is explained that this may in fact return an accessor for the height

I'm not sure I get that. The next sentence mentions that a signal can be returned too.

Reply:
> it doesn't expect any specific unit, just a number, what it means is up to the user

This part of information might be unclear at that point.